### DISCUSSION TITLE
`aligned_depth_to_color` topic is scaled in the z and y axis (x and y in image) ruining calibration to other cameras.

### DISCUSSION THREAD

**Discussion by epanov** [realsenseai/realsense-ros#3198] [Q&A]
In the picture below the raw depth and the aligned to rgb depth is shown on top of a point cloud from an rs455 using the depth_aligned topic. 
![image](https://github.com/user-attachments/assets/7af7903f-05b3-4270-81f5-721a4185670f)
The picture below shows the `aligned_depth_to_color` topic on top of the rs455 point cloud. Note the stretching of this point cloud.
![image](https://github.com/user-attachments/assets/cbddd0db-0953-45f1-b16b-c4d4b2222ace)

Note how the depth, x axis in workspace, z axis in realsense image, is accurate. This to me implies that the xdpi and ydpi may be incorrect. 
Looking at the ros2 param.s they are set to default since `ros2 param get /camera/hand_realsense .hand_realsense.color.image_raw.tiff.xdpi` and `ydpi` return -1.

For now I will use the depth/image_rect_raw topic with intensity, but it would be ideal to have an accurate rgbd image.

EDIT:
Added context for second image.
Added more accurate axis names.

**Comment by MartyG-RealSense** [comment]
Hi @epanov  RealSense cameras use Y for the vertical axis and Z for the forward axis.  The ROS standard is X-forward, Y-left, Z-up. 
 However, the RealSense ROS wrapper should automatically convert the RealSense coordinate system so that the output is compatible with the ROS standard.

I note that your Fixed Frame in RViz is set to **Ethernet0/gpe**.  When visualizing a RealSense pointcloud in RViz, Fixed Frame will more commonly be set to **camera_link**

If using camera_link is not possible for you and you need to change the axes, you could explore the possibility of using a **static transform** or alternatively, use the advice in the link below to reconfigure the position / orientation of axes in RViz.

https://answers.ros.org/question/339109/how-to-change-the-cameras-view-in-rviz-in-the-begin/

**Reply by epanov** [reply]
Pardon my mixing of axis's.
My TF tree set such as to map the camera_link frames to the real world. Note how changing from the `aligned_depth_to_color` causes the change. This is not an issue with calibration or TF trees... I have update the original post for more details.

**Comment by epanov** [comment]
After coming back to my work and restarting the nodes it appears that things work now. Will close thread. Apologies for the false alarm.

**Comment by MartyG-RealSense** [comment]
It's no trouble at all.  I'm pleased to hear that it's working for you. Thanks very much for the update!
